### PR TITLE
Take `name` out of DnaDefHash

### LINKED ISSUE #2038: [BUG] Cloned cell's 'name' affects its DNA hash
**Describe the bug**
Giving a cloned cell a 'name' changes it's DNA hash, as does giving two cloned cells different names. 

I might be misunderstanding the purpose of the cell 'name'. I thought that only the cell 'modifiers' should affect the clone's DNA hash, while the cell 'name' is just a label to make it easier to work with.

**Expected behavior**
Two cells cloned with identical modifiers that are cloned from the same original cell should have identical DNA hashes.

**System information:**
 - OS: Ubuntu 22.04
 - Browser: n/a
 - Holochain and HDK Version: 0.1.3

**Steps to reproduce**
minimum reproduction: https://github.com/mattyg/cell-name-dna-hash-bug-reproduction
run the tryorama tests with `npm run test`

### PR DESCRIPTION
**Pull request by maackle**
Fixes #2038

### Summary

Oops! Despite having talked about it extensively and affirming that the name doesn't affect the DNA hash, sure enough, it did. This fixes that.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holochain_zome_types/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## \[Unreleased\]
 
+- A change to `name` in a DnaDef no longer has an effect on the DNA hash [\#2099](https://github.com/holochain/holochain/pull/2099)
```
Comment (jost-s) [inline crates/holochain_zome_types/CHANGELOG.md:10]:
```suggestion
- `name` in DnaDef no longer has an effect on the DNA hash [\#2099](https://github.com/holochain/holochain/pull/2099)
```

File: crates/holochain/src/conductor/conductor/tests.rs [hunk 2]
```diff
@@ -715,6 +715,47 @@ async fn test_enable_disable_enable_clone_cell() {
...
+            app_id: app_id2.clone(),
+            role_name: role_name2.clone(),
+            modifiers: DnaModifiersOpt::default().with_network_seed("new seed".into()),
+            membrane_proof: None,
+            name: Some("name2".to_string()),
+        })
+        .await
+        .unwrap();
+
+    assert_eq!(clone1.cell_id.dna_hash(), clone2.cell_id.dna_hash());
```
Comment (jost-s) [inline crates/holochain/src/conductor/conductor/tests.rs:756]: No provided name should also not affect the DNA hash.

I would find it clearer if the test used only 1 agent and 1 app and 1 role.
```suggestion
    let clone2 = conductor
        .create_clone_cell(CreateCloneCellPayload {
            app_id: app_id2.clone(),
            role_name: role_name2.clone(),
            modifiers: DnaModifiersOpt::default().with_network_seed("new seed".into()),
            membrane_proof: None,
            name: Some("name2".to_string()),
        })
        .await
        .unwrap();
        
    let clone3 = conductor
        .create_clone_cell(CreateCloneCellPayload {
            app_id: app_id2.clone(),
            role_name: role_name2.clone(),
            modifiers: DnaModifiersOpt::default().with_network_seed("new seed 2".into()),
            membrane_proof: None,
            name: None,
        })
        .await
        .unwrap();

    assert_eq!(clone1.cell_id.dna_hash(), clone2.cell_id.dna_hash());
    assert_eq!(clone1.cell_id.dna_hash(), clone3.cell_id.dna_hash());
```

**Review by jost-s — APPROVED** [review]

File: crates/holochain/src/conductor/conductor/tests.rs (hunk 2 above)
Comment (jost-s) [reply crates/holochain/src/conductor/conductor/tests.rs]: > I would find it clearer if the test used only 1 agent and 1 app and 1 role.

Wait, that's what this PR is fixing. You can't create clone cells with the same modifiers and just different names any longer.

Comment (maackle) [reply crates/holochain/src/conductor/conductor/tests.rs]: I expanded it to 3 cells to test that name = None, Some(X), and Some(Y) all yield the same hash

Comment (jost-s) [reply crates/holochain/src/conductor/conductor/tests.rs]: Yes yes, I realized that is the only way to test.

**Pull request merged**